### PR TITLE
Add missing NMEA patterns

### PR DESCRIPTION
**Pull request by m-kuhn**
Support for:

- GNVTG
- GNGST
- GNHDT

## Description

Variables like HDOP etc. are missing depending on the positioning device in use.

On a more general note, we should likely ignore the second character `N` or `P` in the matching, it is part of the talker ID and ignored by the nmea library as well. It might even be better to completely ignore the first two characters `GN` given the talker id list on https://gpsd.gitlab.io/gpsd/NMEA.html#_gga_global_positioning_system_fix_data , there are some that don't have `G` as first character.

### CODE REVIEW

File: src/core/gps/qgsnmeaconnection.cpp [hunk 1]
```diff
@@ -115,7 +115,7 @@ void QgsNmeaConnection::processStringBuffer()
           mStatus = GPSDataReceived;
           QgsDebugMsgLevel( QStringLiteral( "*******************GPS data received****************" ), 2 );
         }
-        else if ( substring.startsWith( QLatin1String( "$GPVTG" ) ) )
+        else if ( substring.startsWith( QLatin1String( "$GPVTG" ) ) || substring.startsWith( QLatin1String( "$GNVTG" ) ) )
```
Comment (nyalldawson) [inline src/core/gps/qgsnmeaconnection.cpp:118]: This won't have any impact unless you also modify `nmea_parse_GPVTG`. See e.g. 299c61292b58602c4d6f59628b3bafd6125abd19 for an example on what you'd need to change in parse.c here!

Comment (m-kuhn) [reply src/core/gps/qgsnmeaconnection.cpp:118]: Thanks @nyalldawson 
I have only read the G?GGA parser and extrapolated that the others would have been patched also.
As noted in earlier comments, not completely convinced for failing on `type not in 'PN'` but kept the style for consistency.

File: external/nmea/parse.c [hunk 2]
```diff
@@ -283,19 +283,28 @@ int nmea_parse_GPGST( const char *buff, int buff_sz, nmeaGPGST *pack )
 
   nmea_trace_buff( buff, buff_sz );
 
+  char type;
+
   if ( 8 != nmea_scanf( buff, buff_sz,
```
Comment (nyalldawson) [inline external/nmea/parse.c:288]: you need to bump these expected result numbers too... have you tested this?

Comment (m-kuhn) [reply external/nmea/parse.c:288]: not at all. I don't have any means to test this, so the idea was to ship and let a user with the possibilities test the nightlies. Sorry for not being more helpful.

Comment (m-kuhn) [reply external/nmea/parse.c:288]: ... I hope to be able to work on unit tests for this as part of something that could hit our pipeline.

Comment (nyalldawson) [reply external/nmea/parse.c:288]: :smile: (just don't let @roya0045 hear that... shhhhhh :shushing_face: )

**Review by nyalldawson — APPROVED** [review]

File: external/nmea/parse.c (hunk 2 above)
Comment (roya0045) [reply external/nmea/parse.c]: @nyalldawson I'm not sure if I shoul be insulted by this, but its fine. We can all be bumblefucks at times.

Comment (m-kuhn) [reply external/nmea/parse.c]: I think it was a friendly pat to mostly remind me of shortcomings I've imputed to others 😜

Comment (nyalldawson) [reply external/nmea/parse.c]: Don't be... I'm insulting @m-kuhn and calling him out for hypocrisy, not insulting you! 😁 (And @m-kuhn knows me well enough to know that 90% of what I say is firmly tongue in cheek!!)

Comment (roya0045) [reply external/nmea/parse.c]: This is what I was thinking. No offence was taken ;)  I am also well aware of my shortcoming and always find refreshing to see someone else do things blindly from time to time.

**Comment by m-kuhn** [comment]
I received confirmation that it's functional, let's backport !

**Pull request merged**